In bzk run, Wait for mongo to start before starting server

Sometimes, when running `bzk run`, when mongo takes some time to start, the server will try to start but fail because mongo is not yet reachable.

```
panic: no reachable servers

goroutine 1 [running]:
github.com/bazooka-ci/bazooka/commons/mongo.NewConnector(0x870110)
	/go/src/github.com/bazooka-ci/bazooka/commons/mongo/connector.go:47 +0x122
main.main()
	/go/src/github.com/bazooka-ci/bazooka/server/main.go:41 +0x4c6

goroutine 5 [sleep]:
gopkg.in/mgo%2ev2.(*mongoCluster).syncServersLoop(0xc20800c600)
	/go/src/gopkg.in/mgo.v2/cluster.go:383 +0x479
created by gopkg.in/mgo%2ev2.newCluster
	/go/src/gopkg.in/mgo.v2/cluster.go:76 +0x20b

goroutine 16 [sleep]:
gopkg.in/mgo%2ev2.(*mongoServer).pinger(0xc208070000, 0x1)
	/go/src/gopkg.in/mgo.v2/server.go:297 +0x169
created by gopkg.in/mgo%2ev2.newServer
	/go/src/gopkg.in/mgo.v2/server.go:90 +0x196

goroutine 18 [sleep]:
gopkg.in/mgo%2ev2.(*mongoServer).pinger(0xc2080700e0, 0x1)
	/go/src/gopkg.in/mgo.v2/server.go:297 +0x169
created by gopkg.in/mgo%2ev2.newServer
	/go/src/gopkg.in/mgo.v2/server.go:90 +0x196
```

Running `bzk run` again starts the server

We should wait a little between mongo and server to avoid this problem

--- a/commons/net.go
+++ b/commons/net.go
@@ -1,0 +1,24 @@
+package bazooka
+
+import (
+	"fmt"
+	"net"
+	"time"
+)
+
+func WaitForTcpConnection(host, port string, retryEvery, timeout time.Duration) error {
+	url := fmt.Sprintf("%s:%s", host, port)
+	giveUp := time.After(timeout)
+	for {
+		select {
+		case <-time.After(retryEvery):
+			conn, err := net.DialTimeout("tcp", url, 100*time.Millisecond)
+			if err == nil {
+				conn.Close()
+				return nil
+			}
+		case <-giveUp:
+			return fmt.Errorf("Coudln't establish a connection to %s:%s after %v", host, port, timeout)
+		}
+	}
+}

--- a/server/main.go
+++ b/server/main.go
@@ -4,6 +4,9 @@ import (
 	"fmt"
 	"net/http"
 	"os"
+	"time"
+
+	lib "github.com/bazooka-ci/bazooka/commons"
 
 	log "github.com/Sirupsen/logrus"
 	bzklog "github.com/bazooka-ci/bazooka/commons/logs"
@@ -38,6 +41,10 @@ func main() {
 	}
 
 	// Configure Mongo
+	if err := lib.WaitForTcpConnection(env[BazookaEnvMongoAddr], env[BazookaEnvMongoPort], 100*time.Millisecond, 5*time.Second); err != nil {
+		log.Fatalf("Cannot connect to the database: %v", err)
+	}
+
 	connector := mongo.NewConnector()
 	defer connector.Close()
 

